pandas/tests/indexes/datetimes/test_arithmetic.py

```diff
@@ -363,6 +363,48 @@ def test_datetimeindex_sub_timestamp_overflow(self):
             with pytest.raises(OverflowError):
                 dtimin - variant
 
+    @pytest.mark.parametrize('box', [np.array, pd.Index])
+    def test_dti_add_offset_array(self, tz, box):
+        dti = pd.date_range('2017-01-01', periods=2, tz=tz)
+        # TODO: check that `name` propogates correctly
+        other = box([pd.offsets.MonthEnd(), pd.offsets.Day(n=2)])
+        res = dti + other
+        expected = DatetimeIndex([dti[n] + other[n] for n in range(len(dti))],
+                                 name=dti.name, freq='infer')
+        tm.assert_index_equal(res, expected)
+
+        res2 = other + dti
+        tm.assert_index_equal(res2, expected)
+
+    @pytest.mark.parametrize('box', [np.array, pd.Index])
+    def test_dti_sub_offset_array(self, tz, box):
+        # GH#18824
+        dti = pd.date_range('2017-01-01', periods=2, tz=tz)
+        other = box([pd.offsets.MonthEnd(), pd.offsets.Day(n=2)])
+        res = dti - other
+        expected = DatetimeIndex([dti[n] - other[n] for n in range(len(dti))],
+                                 name=dti.name, freq='infer')
+        tm.assert_index_equal(res, expected)
+
+    def test_dti_with_offset_series(self, tz):
```

Comment:
can you parametrize this with name (None, same name, other name) to make sure they are propagated correctly.

Reply:
Good call.  Looks like Series.op(Index) was always taking on the name of the Series.